docs_src/design/adr/system-management/001-docker-image-guidelines.md

```diff
@@ -0,0 +1,117 @@
+# Docker image guidelines
+
+## Status
+
+Pending
+
+## Context
+
+When deploying the EdgeX Docker containers some security measures are recommended to ensure the integrity of the software stack.
+
+## Decision
+
+When deploying Docker images, the following flags should be set for heightened security.
+
+- To avoid escalation of privileges each docker container should use the `no-new-privileges` option in their Docker compose file (example below). More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#security-configuration). This follows Rule #4 for Docker security found [here](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-4-add-no-new-privileges-flag).
+
+```docker
+security_opt:
+      - "no-new-privileges:true"
+```
+
+- To further prevent privilege escalation attacks the user should be set for the docker container using the `--user=<userid>` or `-u=<userid>` option in their Docker compose file (example below). More details about this flag can be found [here](https://docs.docker.com/engine/reference/run/#user). This follows Rule #2 for Docker security found [here](https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-2-set-a-user).
+
+```docker
+services:
+  device-virtual:
+    image: ${REPOSITORY}/docker-device-virtual-go${ARCH}:${DEVICE_VIRTUAL_VERSION}
+    user: 4000:4000 # user option using an unprivileged user
```

Comment:
Are there any guidelines as to what UID ranges should be used? On many distros system user UIDs (i.e. used by services) range from 100-999, and regular user UIDs 1000+.

Reply:
Docker user NS remapping, when enabled, usually uses a very high user range.

I am a proponent of giving each service its own UID (maybe == its port number) so that the secret store tokens can be individually ACLed.